tensorboard/plugins/hparams/backend_context_test.py

```diff
@@ -358,6 +358,55 @@ def test_experiment_without_experiment_tag_many_distinct_values(self):
         _canonicalize_experiment(actual_exp)
         self.assertProtoEquals(expected_exp, actual_exp)
 
+    def test_experiment_with_bool_types(self):
+        self.session_1_start_info_ = """
+            hparams:[
+              {key: 'batch_size' value: {bool_value: true}}
+            ]
+        """
+        self.session_2_start_info_ = """
+            hparams:[
+              {key: 'batch_size' value: {bool_value: false}}
+            ]
+        """
+        self.session_3_start_info_ = """
+            hparams:[
+            ]
+        """
+        expected_exp = """
+            hparam_infos: {
+              name: 'batch_size'
+              type: DATA_TYPE_BOOL
+              domain_discrete: {
+                values: [{bool_value: true}, {bool_value: false}]
+              }
+            }
+            metric_infos: {
+              name: {group: '', tag: 'accuracy'}
+            }
+            metric_infos: {
+              name: {group: '', tag: 'loss'}
+            }
+            metric_infos: {
+              name: {group: 'eval', tag: 'loss'}
+            }
+            metric_infos: {
+              name: {group: 'train', tag: 'loss'}
+            }
+        """
+        ctxt = backend_context.Context(
+            self._mock_tb_context, max_domain_discrete_len=1
```

Comment:
Probably clearer not to specify max_domain_discrete_len since it has no impact on the result.